module/apmstreadwayamqp/wrap_test.go

```diff
@@ -0,0 +1,50 @@
+package apmstreadwayamqp
+
+import (
+	"context"
+	"github.com/streadway/amqp"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.elastic.co/apm/v2"
+	"testing"
+)
+
+func TestWrappedChannel_Publish(t *testing.T) {
+	ctx := context.Background()
+	tx := apm.DefaultTracer().StartTransaction("name", "type")
+	require.NotNil(t, tx)
+	ctx = apm.ContextWithTransaction(ctx, tx)
+	initialHeaders := map[string]interface{}{
+		"ela":   "e",
+		"stic":  "l",
+		"stack": "k",
+	}
+
+	// Since in Go maps are always passed by reference, we must copy the initial map
+	copyInitialMap := make(map[string]interface{})
+	for key, value := range initialHeaders {
+		copyInitialMap[key] = value
+	}
+
+	key := "key"
+	exch := "exchange"
+	msg := amqp.Publishing{
+		Headers: initialHeaders,
+	}
+
+	ch := amqp.Channel{}
+	wrCh := WrapChannel(&ch).WithContext(ctx)
+
+	defer func() {
```

Comment:
This is a bummer. We could perhaps use https://github.com/valinurovam/garagemq to run an in-memory AMQP server. I think what you have here is OK for now, but could you please also check the spans created?

Rather than using `apm.DefaultTracer` above, you should use `apmtest.WithTransaction` like in https://github.com/elastic/apm-agent-go/blob/b6f4f158b8f8d8d0c759deb4cd45811439ce70d4/module/apmhttp/client_test.go#L233. The function provided to WithTransaction is given a context that contains at transaction. WithTransaction returns any events produced during that function call.